unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarLOD/AvatarsLODController.cs

```diff
@@ -73,47 +79,72 @@ internal void UpdateLODsBillboard()
             }
         }
 
-        internal void UpdateAllLODs()
+        internal void UpdateAllLODs(int maxAvatars = DataStore.DataStore_AvatarsLOD.DEFAULT_MAX_AVATAR, int maxImpostors = DataStore.DataStore_AvatarsLOD.DEFAULT_MAX_IMPOSTORS)
         {
-            SortedList<float, IAvatarLODController> closeDistanceAvatars = new SortedList<float, IAvatarLODController>();
-            foreach (var avatarKVP in lodControllers)
-            {
-                var lodController = avatarKVP.Value;
-                var position = otherPlayers[avatarKVP.Key].worldPosition;
-                float distanceToPlayer = Vector3.Distance(CommonScriptableObjects.playerUnityPosition.Get(), position);
-                bool isInLODDistance = distanceToPlayer >= DataStore.i.avatarsLOD.LODDistance.Get();
+            int avatarsCount = 0; //Full Avatar + Simple Avatar
+            int impostorCount = 0; //Impostor
 
-                if (isInLODDistance)
+            //Cache .Get to boost performance. Also use squared data to boost distance comparison
+            float lodDistance = LODDistance.Get() * LODDistance.Get();
+            float squaredSimpleAvatarDistance = simpleAvatarDistance.Get() * simpleAvatarDistance.Get();
+            Vector3 ownPlayerPosition = CommonScriptableObjects.playerUnityPosition.Get();
+
+            (IAvatarLODController lodController, float sqrtDistance)[] data = lodControllers.Values.Select(x => (lodController: x, sqrtDistance: SqrtDistanceToOwnPlayer(x.player, ownPlayerPosition))).ToArray();
+            Array.Sort(data, (x, y) => x.sqrtDistance.CompareTo(y.sqrtDistance));
```

Comment:
Can we improve the style of this big one-liner? It's pretty unreadable in the current state. I suggest to split it into several lines, i.e.:
```
(IAvatarLODController lodController, float sqrtDistance)[] data = null;

data = lodControllers.Values.Select(
    x => (lodController: x, 
              sqrtDistance: SqrtDistanceToOwnPlayer(x.player, ownPlayerPosition))
    ).ToArray();
```
Also, please be very wary of clever coding. Sometimes it's better to use "dumb" code that doesn't use any C# fancy feature, but its more maintainable on the long term.

Reply:
Let me know if you find it more readable now!

I might have some professional deformation from my time working in ASP Net. Core where the use of this features is common and  recommended =(, will try to avoid it!